libsolidity/formal/CHC.cpp

```diff
@@ -1011,27 +1047,30 @@ smtutil::Expression CHC::predicate(Predicate const& _block)
 
 smtutil::Expression CHC::predicate(FunctionCall const& _funCall)
 {
-	/// Used only for internal calls.
-
 	auto const* function = functionCallToDefinition(_funCall);
 	if (!function)
 		return smtutil::Expression(true);
 
+	auto getContractAddressValue = [this](FunctionCall const& _f)
+	{
+		FunctionType const& funType = dynamic_cast<FunctionType const&>(*_f.expression().annotation().type);
+		if (funType.kind() == FunctionType::Kind::Internal)
+			return state().thisAddress();
+		if (MemberAccess const* callBase = dynamic_cast<MemberAccess const*>(&_f.expression()))
+			return expr(callBase->expression());
```

Comment:
I think here you can just return `expr(_f.expression())`, since that's the function type. (without the memberaccess cast I mean)

Reply:
For an expression like `d.f()` or `this.f` I need to get the SMT expression corresponding to `d` or `this`, respectively. Wouldn't what you suggest give me an SMT expression corresponding to `d.f` or `this.f`, respectively?

Reply:
Ah true. I'm not 100% about the unreachable, because of function pointers, but they should be filtered out by `functionCallToDefinition`, so we can keep this part as is for now